### PR TITLE
Feature/ele 191 planning view

### PR DESCRIPTION
**Pull request by gularsson**
Needs https://github.com/ttab/elephant-ui/pull/10

### CODE REVIEW

File: src/components/PlanningTable/RowActions.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        <Button
+          variant="ghost"
+          className="flex h-8 w-8 p-0 data-[state=open]:bg-muted"
+        >
+          <DotsHorizontalIcon className="h-4 w-4" />
+          <span className="sr-only">Open menu</span>
+        </Button>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent align="end" className="w-[160px]">
+        {row.original._source?.['document.meta.core_assignment.rel.deliverable.uuid']
```
Comment (dannelundqvist) [inline src/components/PlanningTable/RowActions.tsx:36]: An empty array of deliverable uuids would enter the loop and output nothing where I would probably expect 'No deliverables' (if that was the intention). In addition, a ternary with long property checks and inline loops inside of rendering makes it difficult to read and understand. Could (or should...) be rewritten as something like:

```
const deliverableUuids = row.original._source?.['document.meta.core_assignment.rel.deliverable.uuid'] || []

return (
  ...
  <DropdownMenuContent align="end" className="w-[160px]">
    {!deliverableUuids.length && 'No deliverables}
    {deliverableUuids.length && deliverableUuids.map(uuid => {
       return <DropdownMenuItem key={uuid>
         <Link ...>
           {uuid}
         </Link>
       </DropdownMenuItem>
    }
  </DropdownMenuContent>
   ...
)
```

File: src/components/PlanningTable/index.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+      <DataTableToolbar table={table} />
+      <div className="rounded-md border">
+        <Table>
+          <TableHeader>
+            {table.getHeaderGroups().map((headerGroup) => (
+              <TableRow key={headerGroup.id}>
+                {headerGroup.headers.map((header) => {
+                  return (
+                    <TableHead key={header.id}>
+                      {header.isPlaceholder
```
Comment (dannelundqvist) [inline src/components/PlanningTable/index.tsx:81]: Would be easier to read as a oneline, even if a tad bit long.

```
{header.isPlaceholder ? null : flexRender(header.column.columnDef.header, header.getContext())}
```

Comment (dannelundqvist) [reply src/components/PlanningTable/index.tsx:81]: A little bit picky ;)

File: src/components/PlanningTable/index.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                          header.getContext()
+                        )}
+                    </TableHead>
+                  )
+                })}
+              </TableRow>
+            ))}
+          </TableHeader>
+          <TableBody>
+            {table.getRowModel().rows?.length
```
Comment (dannelundqvist) [inline src/components/PlanningTable/index.tsx:94]: Large ternaries makes it difficult to follow the logic, Should be refactored into it's own helper render functions in the same file.

**Review by dannelundqvist — CHANGES_REQUESTED** [review]
Lots of work done. Great job with the typings! Most comments are about ternaries making the rendering logic hard to follow in many cases.

File: src/components/PlanningTable/ColumnHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    <div className={cn('flex items-center space-x-2', className)}>
+      <DropdownMenu>
+        <DropdownMenuTrigger asChild>
+          <Button
+            variant="ghost"
+            size="sm"
+            className="-ml-3 h-8 data-[state=open]:bg-accent"
+          >
+            <span>{title}</span>
+            {column.getIsSorted() === 'desc'
```
[the remark this answers is not in the document]
Comment (gularsson) [reply src/components/PlanningTable/ColumnHeader.tsx:44]: Great, will use this :+1:

File: src/components/PlanningTable/Pagination.tsx
[the remark this answers is not in the document]
Comment (gularsson) [reply src/components/PlanningTable/Pagination.tsx:1]: Good idea, now it's tied to react-table functions, but you could probably make i generic for search params and/or generic react-table.

File: src/components/PlanningTable/index.tsx (hunk 2 above)
Comment (gularsson) [reply src/components/PlanningTable/index.tsx:81]: I'll think I'll leave this, it requires a change/ignore of eslint rules to be able to keep it on one line.

File: src/components/PlanningTable/RowActions.tsx (hunk 1 above)
Comment (gularsson) [reply src/components/PlanningTable/RowActions.tsx:36]: Good catch!
